1_wqp_pull/src/wqp_inventory.R

```diff
@@ -71,86 +71,113 @@ inventory_wqp <- function(state_id, wqp_pull_params) {
-    
-    # split counties in half and make a 2-part call
-    split <- ceiling(nrow(state_counties)/2)
-    
-    # first half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
-    temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # second half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
-    temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # combine outputs
-    temp_dat <- list(time = temp1$time + temp2$time,
-                     nrow = nrow(temp1$out) + nrow(temp2$out),
-                     out = bind_rows(temp1$out, temp2$out))
-  }
-  
+  # if(!'list' %in% class(temp_dat)) {
+  #   message('State call failed, calling by 1/2 of counties at a time.')
+  #   state_counties <- filter(dataRetrieval::countyCd, STATE %in% gsub('US:', '', state_id))
+  #
+  #   # split counties in half and make a 2-part call
+  #   split <- ceiling(nrow(state_counties)/2)
+  #
+  #   # first half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
+  #   temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # second half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
+  #   temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # combine outputs
+  #   temp_dat <- list(time = temp1$time + temp2$time,
+  #                    nrow = nrow(temp1$out) + nrow(temp2$out),
+  #                    out = bind_rows(temp1$out, temp2$out))
+  # }
+
   # summarize what was retrieved and how long it took
   message('Retrieved ', temp_dat$nrow, ' rows of data in ', ceiling(temp_dat$time), ' seconds.')
-  
+
   return(temp_dat$out)
 }
 combine_inventory <- function(ind_file, ...) {
   dat_out <- bind_rows(...)
-  
+
   # keep columns of interest, put back lat/long
-  # keeping lat/long here in case there was a situation where you'd want to 
+  # keeping lat/long here in case there was a situation where you'd want to
   # take stock/map sites before pulling data
   dat_out <- dat_out %>%
-    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName, 
-           StateName, CountyName, HUCEightDigitCode, latitude = lat, longitude = lon, resultCount)
+    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName,
+           StateName, CountyName, HUCEightDigitCode, latitude = lat,
+           longitude = lon, resultCount) %>%
+    group_by(MonitoringLocationIdentifier) %>%
+    mutate(resultCount = sum(resultCount)) %>%
+    distinct(MonitoringLocationIdentifier)
```

Comment:
What does `dat_out` look like? It should have all of the columns listed in `select`, but I think it won't because you're only keeping `MonitoringLocationIdentifier` column by calling it out in your `distinct()`. Distinct should not specify any arguments so that it is looking for unique rows accross _all_ columns. After you make this change, please verify that `nrow(dat_out) == length(unique(dat_out$MonitoringLocationIdentifier))`.

Reply:
You're correct, `dat_out` only had one variable `MonitoringLocationIdentifier` . 
Since I didn't specify any arguments in `distinct()`, `dat_out` has 9 variables now and I checked the equality of `nrow(dat_out) == length(unique(dat_out$MonitoringLocationIdentifier))` . 

```
**R results:**
Browse[3]> nrow(dat_out) == length(unique(dat_out$MonitoringLocationIdentifier))
[1] TRUE
Browse[2]> names(dat_out)
[1] "OrganizationIdentifier"             "MonitoringLocationIdentifier"      
[3] "ResolvedMonitoringLocationTypeName" "StateName"                         
[5] "CountyName"                         "HUCEightDigitCode"                 
[7] "latitude"                           "longitude"                         
[9] "resultCount"              
```

![image](https://user-images.githubusercontent.com/65974907/114731821-a85e0d00-9d07-11eb-8213-790e31f66e76.png)
![image](https://user-images.githubusercontent.com/65974907/114731877-b318a200-9d07-11eb-994a-49c80f3ae24c.png)